sailfish/qml/EntryDelegate.qml

```diff
@@ -516,4 +516,65 @@ ListItem {
             }
         }
     }
+
+    drag.target: box
+    drag.axis: Drag.XAxis
+    drag.minimumX: -width
+    drag.maximumX: 0
+
+    drag.onActiveChanged: {
+        if (!drag.active) {
+            if (box.x < -width / 3) {
+                state = "markRead"
+            } else {
+                state = "default"
+            }
+        }
+    }
+
+    state: "default"
+
+    states: [
+        State {
+            name: "default"
+        },
+        State {
+            name: "dragging"
+            when: drag.active
+        },
+        State {
+            name: "markRead"
+        }
+    ]
+
+    transitions: [
+        Transition {
+            from: "dragging"
+            to: "default"
+            NumberAnimation {
+                target: box
+                properties: "x"
+                to: 0
+                duration: 200
+            }
+        },
+        Transition {
+            from: "dragging"
+            to: "markRead"
+            SequentialAnimation {
+                ScriptAction {
+                    script: {
+                        root.markedAsRead();
```

Comment:
What do you think about marking as unread as well? I think it would be useful.

```
if (root.read) {
  root.markedAsUnread()
} else {
  root.markedAsRead()
  root.expanded = false
}
```

Reply:
> What do you think about marking as unread as well? I think it would be useful.

Yes, makes perfect sense. That was my first approach before I dismiss it for a reason I forgot.
I modified my commit to reflect this.